packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema-v2.ts

```diff
@@ -276,19 +278,17 @@ const getRDSLayerMapping = async (): Promise<RDSLayerMapping> => {
   return {};
 };
 
-const isSqlLambdaVpcConfigRequired = async (
-  context: $TSContext,
-  secretsKey: string,
-  engine: ImportedRDSType,
-): Promise<VpcConfig | undefined> => {
-  const { secrets } = await getConnectionSecrets(context, secretsKey, engine);
-  const isDBPublic = await testDatabaseConnection(secrets);
-  if (isDBPublic) {
-    // No need to deploy the SQL Lambda in VPC if the DB is public
+const isSqlLambdaVpcConfigRequired = async (context: $TSContext): Promise<VpcSubnetConfig | undefined> => {
+  // If the database is in VPC, we will use the same VPC configuration for the SQL lambda.
+  // Customers are required to add inbound rule for port 443 from the private subnet in the Security Group.
```

Comment:
Since we are now using VPC endpoints specific to the subnet, I didn't have to add any inbound rule. But with my setup, my subnets are all connected (means all traffic allowed from security group).

I will review this again with few more setups and update the comments as required.